### PR TITLE
Model endpoints refactoring

### PR DESCRIPTION
**Pull request by haiodo**
Signed-off-by: Andrey Sobolev <haiodo@gmail.com>

Refactor model Endpoint and workspace registry to be better testable and usable.

## Description
Refactor model Endpoint and workspace registry to be better testable and usable.

## Motivation and Context
Required for next changes related to NSM healing.

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by haiodo** [comment]
@nickolaev @edwarnicke  Please take a look, it is extracted part of https://github.com/networkservicemesh/networkservicemesh/pull/795 related to Endpoint and workspace management changes.

File: controlplane/pkg/model/model.go [hunk 1]
```diff
@@ -22,14 +22,28 @@ type Dataplane struct {
...
+	ClientConnection_Healing    ClientConnectionState   = 2
+	ClientConnection_Closing    ClientConnectionState   = 3
+	ClientConnection_Closed     ClientConnectionState = 4
+)
+
+type DataplaneState int8
+
+const (
+	DataplaneState_None  = 0 // In case dataplane is not yet configured for connection
+	DataplaneState_Ready = 1 // In case dataplane is configured for connection.
```
Comment (nickolaev) [inline controlplane/pkg/model/model.go]: SHould these be of type `DataplaneState`?

File: controlplane/pkg/nsmd/serviceregistry.go [hunk 2]
```diff
@@ -69,15 +69,15 @@ func (impl *nsmdServiceRegistry) NewWorkspaceProvider() serviceregistry.Workspac
 	return NewDefaultWorkspaceProvider()
 }
 
-func (impl *nsmdServiceRegistry) RemoteNetworkServiceClient(nsm *registry.NetworkServiceManager) (remote_networkservice.NetworkServiceClient, *grpc.ClientConn, error) {
-	err := tools.WaitForPortAvailable(context.Background(), "tcp", nsm.GetUrl(), 1*time.Second)
+func (impl *nsmdServiceRegistry) RemoteNetworkServiceClient(ctx context.Context, nsm *registry.NetworkServiceManager) (remote_networkservice.NetworkServiceClient, *grpc.ClientConn, error) {
+	err := tools.WaitForPortAvailable(ctx, "tcp", nsm.GetUrl(), 1*time.Second)
```
Comment (nickolaev) [inline controlplane/pkg/nsmd/serviceregistry.go]: We should have ca entralised way to define these constants. Even better if we cna find some algorithm to calculate them.

File: controlplane/pkg/model/model.go (hunk 1 above)
Comment (haiodo) [reply controlplane/pkg/model/model.go]: Yes, Thanks.

File: controlplane/pkg/nsmd/serviceregistry.go (hunk 2 above)
Comment (haiodo) [reply controlplane/pkg/nsmd/serviceregistry.go]: I've added TODO for WaitForPortAvailable to refactor it.

**Pull request merged**